# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/ipc/PContent.ipdl
# Commit: 0b4595b2c153
# Full Hash: 0b4595b2c1531b83fc24fbf9b0f6ceacfc3506bd
# Author: Nika Layzell <nika@thelayzells.com>
# Date: 2020-04-22 09:35:42
# Regressor Bug: 1580565
# File Overlap Count: 3
# Description:
#   Bug 1580565 - Part 4: Use WindowContext to manage BrowsingContext cached status, r=farre
#   
#   The existing infrastructure which stored cached BrowsingContexts on the
#   BrowsingContextGroup was added before WindowContexts were added, and can cause
#   racing issues with partially discarded trees during process switches.
# ==============================================================================

diff -r 4363e3a3d799 -r 0b4595b2c153 dom/ipc/PContent.ipdl
--- a/dom/ipc/PContent.ipdl	Wed Apr 22 01:37:52 2020 +0000
+++ b/dom/ipc/PContent.ipdl	Wed Apr 22 01:37:55 2020 +0000
@@ -342,6 +342,12 @@
     nsString value;
 };
 
+union SyncedContextInitializer
+{
+    BrowsingContextInitializer;
+    WindowContextInitializer;
+};
+
 /**
  * The PContent protocol is a top-level protocol between the UI process
  * and a content process. There is exactly one PContentParent/PContentChild pair
@@ -826,8 +832,7 @@
 
     // Begin subscribing to a new BrowsingContextGroup, sending down the current
     // value for every individual BrowsingContext.
-    async RegisterBrowsingContextGroup(BrowsingContextInitializer[] aInits,
-                                       WindowContextInitializer[] aWindowInits);
+    async RegisterBrowsingContextGroup(SyncedContextInitializer[] aInits);
 
 #if defined(MOZ_SANDBOX) && defined(MOZ_DEBUG) && defined(ENABLE_TESTS)
     // Initialize top-level actor for testing content process sandbox.
@@ -1594,18 +1599,6 @@
      */
     async DetachBrowsingContext(uint64_t aContextId) returns (bool unused);
 
-    /**
-     * Removes all of 'aContext'\'s children, and caches them in the
-     * BrowsingContextGroup.
-     */
-    async CacheBrowsingContextChildren(MaybeDiscardedBrowsingContext aContext);
-
-    /**
-     * Re-attach all BrowsingContexts in a 'aContext'.
-     */
-    async RestoreBrowsingContextChildren(MaybeDiscardedBrowsingContext aContext,
-                                         MaybeDiscardedBrowsingContext[] aChildren);
-
     async WindowClose(MaybeDiscardedBrowsingContext aContext,
                       bool aTrustedCaller);
     async WindowFocus(MaybeDiscardedBrowsingContext aContext,